yaksh/views.py

```diff
@@ -619,7 +624,8 @@ def enroll(request, course_id, user_id=None, was_rejected=False):
     ci = RequestContext(request)
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = Course.objects.filter(Q(creator=user)|Q(teachers=user),
```

Comment:
Here, can we get the code as per the id.
And then call course.is_creator or course.is_teacher?
We can do this for 
if quiz.course.creator != user:
as well.